### PR TITLE
Implement Industry::updateProducedCargo

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/Industry.cpp [hunk 1]
```diff
@@ -265,4 +266,57 @@ namespace OpenLoco
...
+                    bits &= ~(1 << bit);
+                    const auto stationId = static_cast<StationId>((dword << 5) | bit);
+                    const auto* station = StationManager::get(stationId);
+                    if (station->empty())
+                    {
+                        continue;
+                    }
+
+                    const auto& cargoStats = station->cargoStats[cargoType];
+                    if (!(cargoStats.flags & (1 << 1)))
```
Comment (LeftofZen) [inline src/OpenLoco/Industry.cpp:300]: any reason not to write `if (!(cargoStats.flags & 0x10))`

Comment (duncanspumpkin) [reply src/OpenLoco/Industry.cpp:300]: i assume you mean 0x02. Thats how we write the flags before they are added anywhere. Its much easier to find in the codebase (1<<1) than it is to find 0x2 and if there were other flags its much easier to understand (1<<1)|(1<<0) compared to 0x3.

Comment (LeftofZen) [reply src/OpenLoco/Industry.cpp:300]: yeah for some reason i was thinking in binary :| but no worries, was just wondering what convention was

**Comment by LeftofZen** [comment]
I haven't yet opened this function(s) in IDA yet, but currently it looks good to me

**Review by LeftofZen — APPROVED** [review]

File: src/OpenLoco/Industry.cpp [hunk 2]
```diff
@@ -265,4 +266,57 @@ namespace OpenLoco
...
+                    {
+                        continue;
+                    }
+
+                    const auto rating = cargoStats.rating;
+                    for (auto index = 0; index < 4; ++index)
+                    {
+                        if (indStatsStation[index] == StationId::null || indStatsRating[index] < rating)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Industry.cpp:308]: <=

**Pull request merged**